src/CMakeLists.txt

```diff
@@ -842,3 +842,39 @@ if (IS_DIRECTORY "${PROJECT_SOURCE_DIR}/.git")
 endif()
 
 add_subdirectory(script)
+
+# add doxygen target
+find_package(Doxygen)
+if(DOXYGEN_FOUND)
+  set(DOXYGEN_EXCLUDE_PATTERNS blkin 
```

Comment:
maybe this will be easier to maintain if it's opt-in instead of opt-out? we tend to add lots of submodules, and nobody will remember to add them here. but if you're adding comments to a new path and want them to build with 'make doxygen', then you know where to look

what do you think?

Reply:
@cbodley makes sense. will change